ui/src/components/Configuration/ConfigurationHelper.ts

```diff
@@ -0,0 +1,578 @@
+ * @param monthly Monthly Crontab Value
+ * @param weekly Weekly Crontab Value
+ * @returns On what schedule this Crontab is happening.
+ */
+const parseOccuranceDetails = (monthly: number, weekly: number) => {
+  let occuranceDetails = { occurance: 'Daily', occuranceIndex: 0 }
+  occuranceDetails = !isNaN(monthly) ? { occurance: 'Monthly', occuranceIndex: 1 } : occuranceDetails
+  occuranceDetails = !isNaN(weekly) ? { occurance: 'Weekly', occuranceIndex: 2 } : occuranceDetails
+  return occuranceDetails
+}
+
+/**
+ * Last step before saving to the server.
+ * @param dataToUpdate Full Server-Ready, already converted with convertLocalToServer object to be saved to the server.
+ * @returns An object for the server without the additional fields we tacked on for context.
+ */
+const stripOriginalIndexes = (dataToUpdate: Array<ProvisionDServerConfiguration>) => {
+  return dataToUpdate.map((item) => {
+    delete item.originalIndex
+    delete item.currentSort
+    return item
+  })
+}
+
+/**
+ * Validates a Hostname. Can be an IP address or valid hostname.
+ * @param host Hostname
+ * @returns Blank if Valid, Error Message if Not.
+ */
+const validateHost = (host: string) => {
+  let hostError = ''
+  const ipv4 = new RegExp(
+    /^(([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])(:[0-9]+)?$/gim
+  )
+  const isHostValid = ipv4.test(host)
+  if (!isHostValid) {
+    hostError = ErrorStrings.InvalidHostname
+  }
+  return hostError
+}
+
+/**
+ * This function validates the local configuration when a
+ * user selects Save & Close, or when the user types in any field.
+ * If you want the errors to stay on screen until the user clicks Save & Close again
+ * remove the quickUpdate functionality below and it will validate on every keystroke.
+ * @param localItem Local/ProvisionD Configuration Item
+ * @param quickUpdate Just clear the errors when the user starts to type.
+ * @returns If the Local Configuration Item is valid (and therefore ready to close the window)
+ */
+const validateLocalItem = (localItem: LocalConfiguration, quickUpdate = false): LocalErrors => {
+  const errors: LocalErrors = createBlankErrors()
+
+  if (!quickUpdate) {
+    errors.name = validateName(localItem.name)
+    errors.type = validateType(localItem.type.name)
+
+    if (RequsitionTypesUsingHost.includes(localItem.type.name)) {
+      errors.host = validateHost(localItem.host)
+    }
+    if (localItem.type.name === RequisitionTypes.DNS) {
+      errors.zone = validateHost(localItem.zone)
+
+      // Only validate foreign source if it's set.
+      if (localItem.foreignSource) {
+        errors.foreignSource = validateHost(localItem.foreignSource)
+      }
+    }
+    if (localItem.type.name === RequisitionTypes.File) {
+      errors.path = validatePath(localItem.path)
+    }
+    errors.occurance = validateOccurance(localItem.occurance.name)
+  }
+
+  //If any key is set, then we have errors.
+  for (const [_, val] of Object.entries(errors)) {
```

Comment:
getting a lint warning on this one.
maybe `const val of Object.values(errors)` ?

Reply:
Done! 

Normally the '_' tells a linting tool "I know you're gonna complain about this, but I know what I'm doing, trust me" but this other way works too.